src/auxiliary/mpi.jl

```diff
@@ -110,7 +110,7 @@ end
 # care of these situations when allowing to use `ode_norm` as default norm in
 # OrdinaryDiffEq.jl throughout all applications of Trixi.jl.
 recursive_sum_abs2(u::Number) = abs2(u)
-recursive_sum_abs2(u::AbstractArray) = sum(recursive_sum_abs2, u)
+recursive_sum_abs2(u::AbstractArray) = sum(recursive_sum_abs2, u; init=0)
```

Comment:
I believe this should read
```suggestion
recursive_sum_abs2(u::AbstractArray) = sum(recursive_sum_abs2, u; init=zero(eltype(u)))
```
to be type stable (cc @ranocha )

Reply:
Yes, exactly :+1:

Reply:
Yes, you are right. Done.

Reply:
Actually, this doesn't seem to be the optimal solution since `u` can be an array-of-arrays such that `eltype(u)` would be e.g. `SVector{2, Float64}`, but what we want is `Float64`, doesn't it? So we could write `init=zero(eltype(eltype(u))`, but this doesn't seem to be very elegant, either. Any thoughts on that?

Reply:
I made a suggestion [below](https://github.com/trixi-framework/Trixi.jl/pull/1326#discussion_r1071317870) to include this additional information as a comment.

Reply:
Thanks, nice to know that this exists.